src/App.js

```diff
@@ -23,56 +20,47 @@ class App extends Component {
 	}
 
 	componentDidMount = () => {
-		this.setState({ movies: movieData.movies });
-		// fetch from URL _____
-		//.then that parses the response object
-		// a .then that invokes a function in utils.js that cleans up the data to what we want to display
-		// Should set state
-		// Data will reflect:
-		// id
-		// movie
-		// poster_path
-		// backdrop_path
-		// release_date
-		// overview
-		// average_rating
-		// genres
-		// id
-		// name
-		// budget
-		// revenue
-		// runtime
-		// tagline
-	};
-
-	// showSelectedMovieDetails = (id) => {
-	// 	console.log(id);
-	// 	// Should hide banner and cards container
-	// 	// Should show movie details display
-	// };
-
-	hideSelectedMovieDetails = () => {
-		this.setState({ selectedMovie: {} });
+		fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+			.then((response) => {
+				if (response.ok) {
+					return response.json();
+				}
+				throw new Error('[all movies] Something went wrong. Please try again.');
+			})
+			.then((data) => {
+				this.setState({ movies: data.movies });
+			})
+			.catch((error) => this.setState({ error: error.toString() }));
 	};
 
 	setSelectedMovieToState = (id) => {
-		const selectedMovie = this.state.movies.find((movie) => {
-			return movie.id === id;
-		});
-		console.log('id', id);
-		console.log('selectedMovie', selectedMovie);
-		this.setState({ isAMovieSelected: true, selectedMovie: selectedMovie });
+		this.setState({ isAMovieSelected: true });
+		fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/mmovies/${id}`)
+			.then((response) => {
+				if (response.ok) {
+					return response.json();
+				}
+				throw new Error('[for individual movie] Something went wrong. Please try again.');
+			})
+			.then((data) => {
+				if (!this.state.error) {
+					this.setState({ selectedMovie: data.movie });
```

Comment:
Same here - the fetch call gives you LOTS of data - if you're not using it, trim it down before setting to state.